src/backend/test.rs

```diff
@@ -66,6 +66,10 @@ impl TestBackend {
         self.height = height;
     }
 
+    pub fn print_buffer(&self) -> String {
```

Comment:
`print` doesn't seem like the right verb for this (as it would imply sending to stdout) - perhaps `to_string()`? Do you have an idea on what the idiomatic rust name for something like this might be?

I'm concerned however that this function might be modified at some stage in the future (like perhaps better display of the multi-widget overwrites, or removal of the quotation marks / last newline. That could cause anything relying on this (which would generally be in tests) to break. Do you have a use case in mind for this?

I'm assuming that you probably need this for integration tests in your app (if you're using this for unit tests, perhaps take a quick look at #161 and see if that suits your use case at all). Guessing that you could be using this for documentation too?

Reply:
I agree, I wasn't sure on the naming. 

I thought about adding a `Display` implementation for `Buffer`, but I intentionally wanted to have it exclusively in the `test` interface and didn't want to shadow `to_string()` itself. I like your solution in #161, but I will want to rely on `to_test_string()` in that interface.

> I'm concerned however that this function might be modified at some stage in the future (like perhaps better display of the multi-widget overwrites, or removal of the quotation marks / last newline. 

I am OK with that, I don't mind going over the tests in that case.

Reply:
I opted implementing `Display` for `TestBuffer` in the end, and I am quite happy with that. That way the Buffer itself doesn't get a `to_string()` method, but because it stays in the `TestBuffer`, it also stays in the test context.